src/lib.rs

```diff
@@ -271,13 +272,27 @@ where
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
         trace!("{}:{} Stream.poll_next", file!(), line!());
+
+        // The connection has been closed or the critical error has occurred.
+        // We have already returned the error to the user, the `Stream` is unusable,
+        // so we assume that the stream has been "fused".
```

Comment:
You probably also want to implement `futures::stream::FusedStream` then

Reply:
Oh, great suggestion! Thanks. Will update the PR right away.